### PR TITLE
[Sprint: 55] XD-3347 Run all shell integration tests also with enabled security

### PR DESCRIPTION
**Pull request by ghillert**
* Refactor Shell command tests
* Fix security rules for stopping all job executions

Jira: https://jira.spring.io/browse/XD-3347

### CODE REVIEW

**Comment by ericbottard** [comment]
I wonder if this could call be replaced with some JUnit magic, similar to what `@Parameterized` does, _etc._

**Comment by ghillert** [comment]
There should be maybe a way to also do something with Junit rules but the info on this was a bit slim and I spent way too much time just to get the tests pass and fix some related issues.

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractFileSourceAndFileSinkTests.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
  * @author David Turanski
  * @author Franck Marchand
  */
-public class FileSourceAndFileSinkTests extends AbstractStreamIntegrationTest {
+public class AbstractFileSourceAndFileSinkTests extends AbstractStreamIntegrationTest {
 
 	private static final File DEFAULT_IN = new File("/tmp/xd/input");
```
Comment (markpollack) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractFileSourceAndFileSinkTests.java]: This isn't going to work on windows... I guess it was excluded from the build before?
Maybe `new File(System.getProperty("java.io.tmpdir"));`

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractJobCommandsWithHadoopTests.java [hunk 2]
```diff
@@ -45,14 +45,14 @@
  * @author Gary Russell
  */
 @RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(loader = HadoopDelegatingSmartContextLoader.class, classes = JobCommandsWithHadoopTests.EmptyConfig.class)
+@ContextConfiguration(loader = HadoopDelegatingSmartContextLoader.class, classes = AbstractJobCommandsWithHadoopTests.EmptyConfig.class)
 @MiniHadoopCluster
-public class JobCommandsWithHadoopTests extends AbstractJobIntegrationTest {
+public class AbstractJobCommandsWithHadoopTests extends AbstractJobIntegrationTest {
```
Comment (markpollack) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractJobCommandsWithHadoopTests.java]: Why was this made into an abstract class?

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractModuleClasspathTests.java [hunk 3]
```diff
@@ -31,7 +35,7 @@
  * @author Eric Bottard
  * @author David Turanski
  */
-public class ModuleClasspathTests extends AbstractStreamIntegrationTest {
+public class AbstractModuleClasspathTests extends AbstractStreamIntegrationTest {
 	//todo:What is the purpose of this test?
```
Comment (markpollack) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractModuleClasspathTests.java]: yes, that is the purpose... the use of siDslModule?

Comment (ghillert) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractModuleClasspathTests.java]: Removing that line. Was there before :-)

File: spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractJobCommandsWithHadoopTests.java (hunk 2 above)
Comment (ghillert) [inline spring-xd-shell/src/test/java/org/springframework/xd/shell/command/AbstractJobCommandsWithHadoopTests.java]: Looks like I forgot to cover that one - will fix

**Comment by markpollack** [comment]
cherry-picked onto 1.2.x as ea665bddb0694619b4820ef39a5fcea299f0ddef and merged into master as ea665bddb0694619b4820ef39a5fcea299f0ddef

**Pull request closed without merging**